plugins/c8y_log_plugin/src/logfile_request.rs

```diff
@@ -0,0 +1,165 @@
+            CumulocitySupportedOperations::C8yLogFileRequest,
+            failure_reason,
+        )
+        .to_smartrest()
+    }
+}
+/// Reads tedge logs according to `SmartRestLogRequest`.
+///
+/// If needed, logs are concatenated.
+///
+/// Logs are sorted alphanumerically from oldest to newest.
+///
+/// # Examples
+///
+/// ```
+/// let smartrest_obj = SmartRestLogRequest::from_smartrest(
+///     "522,DeviceSerial,syslog,2021-01-01T00:00:00+0200,2021-01-10T00:00:00+0200,,1000",
+/// )
+/// .unwrap();
+///
+/// let log = read_tedge_system_logs(&smartrest_obj, "/var/log/tedge").unwrap();
+/// ```
+pub fn read_tedge_logs(
+    smartrest_obj: &SmartRestLogRequest,
+    plugin_config_path: &PathBuf,
+) -> Result<String, anyhow::Error> {
+    let plugin_config = PluginConfig::new(&plugin_config_path);
+    let mut output = String::new();
+
+    let mut files_to_send = Vec::new();
+    for files in &plugin_config.files {
+        let maybe_file_path = files.path.as_str(); // because it can be a glob pattern
+        let file_type = files.config_type.as_str();
+
+        if !file_type.eq(&smartrest_obj.log_type) {
+            continue;
+        }
+
+        // NOTE: According to the glob documentation paths are yielded in alphabetical order hence re-ordering is no longer required see:
+        // https://github.com/thin-edge/thin-edge.io/blob/0320741b109f50d1b0f7cda44e33dc31ba04902d/plugins/log_request_plugin/src/smartrest.rs#L24
+        dbg!(&maybe_file_path);
+        for entry in glob(maybe_file_path)? {
+            dbg!(&entry);
+            let file_path = entry?;
+            if let Some(dt_from_file) = get_datetime_from_file_path(&file_path) {
+                if !(dt_from_file < smartrest_obj.date_from || dt_from_file > smartrest_obj.date_to)
+                {
+                    files_to_send.push(file_path);
+                }
+            } else {
+                files_to_send.push(file_path);
+            }
+        }
+    }
+    dbg!(&files_to_send);
+
+    // loop sorted vector and push store log file to `output`
+    let mut line_counter: usize = 0;
+    for entry in files_to_send {
+        dbg!("files to read:", &entry);
+        let file_content = std::fs::read_to_string(&entry)?;
+        if file_content.is_empty() {
+            dbg!("content was empty");
+            continue;
+        }
+
+        // adding file header only if line_counter permits more lines to be added
+        match &entry.file_stem().and_then(|f| f.to_str()) {
+            Some(file_name) if line_counter < smartrest_obj.lines => {
+                output.push_str(&format!("filename: {}\n", file_name));
+            }
+            _ => {}
+        }
+
+        // split at new line delimiter ("\n")
+        let mut lines = file_content.lines();
+        while line_counter < smartrest_obj.lines {
+            if let Some(haystack) = lines.next() {
+                if let Some(needle) = &smartrest_obj.needle {
```

Comment:
Smart name but may not be obvious to others. How about
```suggestion
                if let Some(filter_text) = &smartrest_obj.filter_text {
```